File: typescript/packages/bench/src/rules/labelPresenceRule.ts
/** Cardinality fixture: nominal (class, `implements`) implementor of RuleContract. */

import type { RuleContract } from "../contracts/ruleContract";
import type { Dataset } from "../dataset";

export class LabelPresenceRule implements RuleContract {
    readonly key = "label-presence";

    evaluate(data: Dataset): boolean {
        return data.devices.length >= 1;
    }
}

File: rust/crates/atelier-bench/src/rules/label_presence_rule.rs
//! Rule 14/24 (textual half).

use crate::contracts::rule_contract::RuleContract;
use crate::dataset::Dataset;

pub struct LabelPresenceRule;

impl RuleContract for LabelPresenceRule {
    fn evaluate(&self, data: &Dataset) -> bool {
        !data.devices.is_empty()
    }
}
